src/runtime_src/core/pcie/linux/system_linux.cpp

```diff
@@ -131,7 +131,10 @@ get_pcidev(unsigned index, bool is_user) const
     if (index < user_ready_list.size())
       return user_ready_list[index];
 
-    return user_nonready_list[index - user_ready_list.size()];
+    if ((index - user_ready_list.size()) < user_nonready_list.size())
+      return user_nonready_list[index - user_ready_list.size()];
```

Comment:
```suggestion
      return user_nonready_list.at(index - user_ready_list.size());
```
WIth try catch.  Do same for mgmt